src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -229,8 +229,57 @@ chip::Protocols::InteractionModel::Status
 MatterDoorLockClusterServerPreAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath,
                                                        EmberAfAttributeType attributeType, uint16_t size, uint8_t * value)
 {
-    // TODO: Implement attribute changes
-    return chip::Protocols::InteractionModel::Status::Success;
+    chip::Protocols::InteractionModel::Status res;
+
+    switch (attributePath.mAttributeId)
+    {
+    case chip::app::Clusters::DoorLock::Attributes::Language::Id: {
+        char lang[3 + 1] = { 0 };
+        memcpy(lang, &value[1], value[0]);
```

Comment:
This is going to do very bad things if value[0] > 4.  Please check before doing the copy.

If this is relying on the `length="3"` bits in the XML, those are subject to change at the very least.   At bare minimum, a sanity check of our buffer size against `size` is in order, but better to compare to the thing we really care about, which is `value[0]`.

The other option, of course, is to just use a CharSpan as the arg passed to the callee, and then you don't need to worry about the extra buffer or the copying and overrunning it.  We could even extent Span::fromZclString to work with a CharSpan if that would be helpful here....

Reply:
I relied on the check ``VerifyOrReturnError(value.size() <= 3, EMBER_ZCL_STATUS_CONSTRAINT_ERROR);`` and defined buffer in the same manner like in __Accessors.cpp__. But if to call ``emberAfWriteServerAttribute()`` directly, looks like we can pass any string to __language__ attribute and cluster will receive it as is as the length constraint isn't checked by the framework, just min/max.